packages/@aws-cdk/integ-runner/lib/runner/integration-tests.ts

```diff
@@ -37,6 +37,22 @@ export interface IntegTestInfo {
  * Derived information for IntegTests
  */
 export class IntegTest {
+  private static readonly defaultSuffixes = new Map<string, RegExp>([
+    ['javascript', new RegExp(/\.js$/)],
+    // Allow files ending in .ts but not in .d.ts
+    ['typescript', new RegExp(/(?<!\.d)\.ts$/)],
+  ]);
+
+  private static readonly defaultAppCommands = new Map<string, string>([
+    ['javascript', 'node {filePath}'],
+    ['typescript', 'node -r ts-node/register {filePath}'],
+  ]);
+
+  private static getLanguage(fileName: string): string | undefined {
+    const [language] = Array.from(IntegTest.defaultSuffixes.entries()).find(([, regex]) => regex.test(fileName)) ?? [undefined, undefined];
+    return language;
+  }
+
```

Comment:
This feels out of place here. See comment further down for a possible refactoring.